File: cinema-backend/src/main/java/org/example/cinemabackend/movie/application/dto/response/MovieListResponse.java
package org.example.cinemabackend.movie.application.dto.response;

import lombok.Builder;

import java.time.LocalDate;

@Builder
public record MovieListResponse(
        String title,
        String originalTitle,
        Integer duration,
        LocalDate releaseDate,
        String poster,
        FilmMemberResponse director
) {
}

File: cinema-admin-frontend/src/app/movie/dtos/response/movie-list.response.ts
import {FilmMemberResponse} from "./film-member.response";

export class MovieListResponse {
  constructor(
    readonly title: string,
    readonly originalTitle: string,
    readonly duration: string,
    readonly releaseDate: Date,
    readonly poster: string,
    readonly director: FilmMemberResponse,
  ) {
  }
}
